### PR TITLE
Merge salt tile integration

### PR DESCRIPTION
**Pull request by nkronenfeld**
Batch and Spark/Salt-based tile generation for Veldt.

Not all tile types are implemented - the following are still missing:

- frequency_tile
- target_term_count_tile
- target_term_frequency_tile
- top_term_count_tile
- top_term_frequency_tile

Other future work:
- combining multiple layers from the same dataset into a single request - if there are 5 layers, this will make the salt server approx. 5x as fast
- univariate tiles

I've added Phil because he did similar batching work for a side project - I am hoping for his input specifically for the batching portion of this PR.

### CODE REVIEW

**Comment by nkronenfeld** [comment]
From Kevin, through private communications:
Style:
- error messages
    - don't capitalize first letter message unless referring to quoted `VarName`
- batch package names:
    - batch_queue.go to queue.go
- salt package names
    - salt.NewSaltMeta to salt.NewMeta
    - salt_tile.go to tile.go
    - salt_meta to meta.go
- go-style public attribute naming conventions:
    - `Coordinates` to `Coord`
    - `Configuration` to `Config`
    - `Parameters` to `Params`
    - `ResultChannel` to `C`
- Logging:
    - remove internal logging
    - bubble any logged errors up to user level (looks like you already do this?)
Tests:
    - use ginkgo BDD framework for testing like other packages

File: generation/batch/queue.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+		}
+	}
+
+	return queueBatch, current
+}
+
+// processFactoryRequess process all the requests from a given batch for a
+// given factory
+func processFactoryRequests(batch int, factoryID string, factoryRequests []*tileRequestInfo) {
+	Debugf("Processing %d requests for factory %s", len(factoryRequests), factoryID)
```
Comment (phorne-uncharted) [inline generation/batch/queue.go]: I think typical style guide is to not nest ifs when errors occur. Instead, have an if check the condition and end the if block with a return statement. ex:
if !ok {
    ...
    return
}

**Comment by phorne-uncharted** [comment]
It looks like the formatting of some of the code is off. Was "go fmt" run?

File: generation/batch/queue.go (hunk 1 above)
Comment (nkronenfeld) [inline generation/batch/queue.go]: done

**Pull request merged**